### PR TITLE
Analytical solution uniaxial compression and NHK docs

### LINKED ISSUE #608: Document Neo-Hookean model
Considering this strain energy function is:

 ```math
 \Psi(\mathbf{C}) = \frac{\mu}{2}(I_1 -3) + 1/D  (J -1)^2
 ```
then Cosserat is:

 ```math
 \mathbf{S} = 2 \frac{\partial \Psi }{\partial \mathbf{C}} = \mu (I_{3,3}) + \frac{2}{D}(J-1) J C^{-T}
 ```
but in ONSAS we have:

 ```matlab
S       = shear * ( eye(3) - invC ) + bulk * ( J * (J-1)* invC) ;
```

Useful identities:

 ```math
J  = \sqrt(det(\mathbf{C}))
```
 ```math
\frac{\partial J}{\partial \mathbf{C}}  = \frac{J \mathbf{C}^{-T}}{2}  
```
 ```math
\frac{\partial I_1}{\partial \mathbf{C}}  = I_{3,3}  
```

### PR DESCRIPTION
**Pull request by mvanzulli**
Closes #608 thanks to @bbazzano and @jorgepz  contributions.  Indeed, we are now checking that the nominal stress tensor towards $y$ and $z$ are 0.

### CODE REVIEW

**Comment by jorgepz** [comment]
I think that the functions are well derived [here](https://github.com/ONSAS/ONSAS.m/issues/608#issuecomment-1423122311) but re-check it just in case!

**Comment by jorgepz** [comment]
there seems to be an issue here 
image

File: examples/uniaxialCompression/uniaxialCompression.m [hunk 1]
```diff
@@ -19,31 +19,40 @@
...
 %mdP_{xx}( \mu,K ) =
 %md \alpha \left( 
 %md    \mu -  \frac{\mu}{\alpha^2} + \frac{K\beta^2}{\alpha} (\beta^2 \alpha -1) 
 %md \right) = - p
 %md```
+%md and on the other hand, the $P_{yy}$ and $P_{zz}$ components are obtained and equaled to zero:
+%md```math
+%mdP_{yy}( \mu,K ) =
+%md \beta \left( 
+%md    \mu -  \frac{\mu}{\beta^2} + \frac{K} (\alpha^2\beta^2 - \alpha)
```
Comment (jorgepz) [inline examples/uniaxialCompression/uniaxialCompression.m:45]: there is a typo here

Comment (mvanzulli) [reply examples/uniaxialCompression/uniaxialCompression.m]: Done

**Comment by jorgepz** [comment]
@mvanzulli   Is there any reason for increasing E?

File: examples/uniaxialCompression/uniaxialCompression.m [hunk 2]
```diff
@@ -19,31 +19,40 @@
...
+%md```
 %md
 %md## Numerical solution
 %mdBefore defining the structs, the workspace is cleaned, the ONSAS directory is added to the path and scalar geometry and material parameters are defined.
 clear all, close all
 % add path
 addpath( genpath( [ pwd '/../../src'] ) ) ;
 % scalar parameters
-E = 1 ; nu = 0.3 ; p = -5 ; Lx = 2 ; Ly = 1 ; Lz = 1 ;
+E = 4 ; nu = 0.3 ; p = -5 ; Lx = 2 ; Ly = 1 ; Lz = 1 ;
```
Comment (jorgepz) [inline examples/uniaxialCompression/uniaxialCompression.m:55]: related to https://github.com/ONSAS/ONSAS.m/pull/605#issuecomment-1424130077

Comment (mvanzulli) [reply examples/uniaxialCompression/uniaxialCompression.m]: Done

**Pull request merged**